src/main/java/com/cloudbees/hudson/plugins/folder/properties/FolderCredentialsProvider.java

```diff
@@ -119,21 +124,28 @@
     public <C extends Credentials> List<C> getCredentials(@NonNull Class<C> type, @Nullable ItemGroup itemGroup,
                                                           @Nullable Authentication authentication,
                                                           @NonNull List<DomainRequirement> domainRequirements) {
-        if (authentication == null) {
-            authentication = ACL.SYSTEM;
-        }
         List<C> result = new ArrayList<C>();
+        Set<String> ids = new HashSet<String>();
         if (ACL.SYSTEM.equals(authentication)) {
             while (itemGroup != null) {
                 if (itemGroup instanceof AbstractFolder) {
+                    if (!((AbstractFolder) itemGroup).getACL().hasPermission(authentication, USE_ITEM)) {
```

Comment:
Well, you just verified three lines up that `authentication` is equal to `ACL.SYSTEM` so this seems redundant to me.